### DISCUSSION TITLE
ボタンで画像を増やす時

### DISCUSSION THREAD

**Discussion by ShunHigu** [lit-iphone-course/lit-iphone-course#40] [Question-original] [answered]
１　ボタンを押すたびにランダムで画像を出して、さらにその画像を画面上に残しておくコードを書きたいです。

２　３にも書いたサイトを参考にした結果、エラーは起きなかったのですが、全く画像が表示されませんでした。
「let RendaImage=UIView(frame: CGRect(x: xrand, y: yrand, width: 100, height: 100))」
UIViewの使い方もわからないので教えていただきたいです。

３　画像を指で動かすという先の工程のことも調べた結果、押したら画像が増え、さらに画面上に残り、指で動かすコードを書いているサイトがあったので参考にしてみたのですが、あまりうまくいきませんでした。
https://qiita.com/takehilo/items/0ce4d681b009c56d2b1e

４　let RendaImage=UIView(frame: CGRect(x: xrand, y: yrand, width: 100, height: 100))

５　https://github.com/ShunHigu/Original

**Comment by txxxxc** [comment, marked as answer]
全く画像が表示されていない原因としては、初期化したUIViewを親のViewに追加するコードを書いていないからです！
`addSubview`というメソッドを使えば親のViewに要素を追加できるので試してみてくださいー！
具体的にはこんな感じで書くとうまくいくと思いますー！
```
        photonum=photonum+1
        num=Int.random(in: 1...100)
        let xrand=Double.random(in: 40...370)
        let yrand=Double.random(in: 40...450)
        
        let RendaImage=UIView(frame: CGRect(x: xrand, y: yrand, width: 100, height: 100))
　　// viewは透明なので追加できているか確認するために色を付ける
        RendaImage.backgroundColor = .blue
        view.addSubview(RendaImage)

```

画像を指で動かすやつに関しては、参考記事通りに実装すれば多分行けると思うので、もう一回挑戦してみてもらえるとありがたいです！！

**Reply by ShunHigu** [reply]
ありがとうございました！できました！
